### PR TITLE
tests: unifying the core20 nested suite with the core nested suite

### PR DESCRIPTION
**Pull request by sergiocazzolato**
This change includes:
. core20 and core nested suites are unified into core suite. This
avoids tests duplication
. Now the tpm, secboot and kvm variables have defaults which are used
through functions defined in nested.sh
 . By default the uc20 is build with secboot and tpm enabled.
 . By detauld uc16 and uc18 are not using neither secboot nor tpm
 . By default kvm is always used
. To define a different behaviour, the test needs to define the
variable in its environment.

### CODE REVIEW

**Review by bboozzoo — APPROVED** [review]
LGTM

File: tests/nested/core/hotplug/task.yaml [hunk 1]
```diff
@@ -1,5 +1,7 @@
 summary: Create ubuntu core image, install snapd and test hotplug feature
 
+systems: [ubuntu-16.04-64, ubuntu-18.04-64]
```
Comment (anonymouse64) [inline tests/nested/core/hotplug/task.yaml:3]: hmm should we have a TODO:UC20: for enabling this on UC20?

**Review by anonymouse64 — APPROVED** [review]
LGTM, though I think we should add a TODO:UC20: about enabling the hotplug test for UC20. 

This looks much nicer, thanks for merging these two together like this.

File: tests/nested/core/hotplug/task.yaml (hunk 1 above)
Comment (sergiocazzolato) [reply tests/nested/core/hotplug/task.yaml:3]: I'll add that in a following PR, thanks for the review

**Pull request merged**